Core/Utilities/DEncode.py

```diff
@@ -21,24 +21,69 @@
 import datetime
 import os
 
+import functools
 import inspect
 import traceback
+
+from collections import defaultdict
 from pprint import pprint
 
 
 # Setting this environment variable to any value will enable the dump of the debugging
 # call stack
 DIRAC_DEBUG_DENCODE_CALLSTACK = bool(os.environ.get('DIRAC_DEBUG_DENCODE_CALLSTACK', False))
 
+# This global dictionary contains
+# {<method name> : set (<class names)}
+# (a method name can be reused in other classes)
+# DO NOT EDIT BY HAND, use ignoreEncodeWarning decorator
+DENCODE_WARNING_IGNORED_METHODS = defaultdict(set)
+
 # Depth of the stack to look for with inspect
 CONTEXT_DEPTH = 100
 
 
+def ignoreEncodeWarning(meth):
+  """ Decorator to put around method that should not anymore throw warnings
+
+      :warning: do not use around functions
+
+      :warning: for a class method, put it after the @classmethod decorator
+
+      :param meth: decorated method
+  """
+
+  @functools.wraps(meth)
+  def inner(*args, **kwargs):
+    """ Add the method and the class name to the DENCODE_WARNING_IGNORED_METHODS dict """
+
+    # The first parameter in args is "self"
+    # Find out the class Name
+    objInst = args[0]
+    className = objInst.__class__.__name__
+    if className == 'type':  # This happens for class method
+      className = objInst.__name__
+
+    # if the decorated method is an exported method, just remove the 'export_' bit
+    methName = meth.__name__.replace('export_', '')
+
+    # Add the method name and the object name to the dictionary
+    DENCODE_WARNING_IGNORED_METHODS[methName].add(className)
+    return meth(*args, **kwargs)
+
+  return inner
+
+
 def printDebugCallstack(headerMessage):
   """ Prints information about the current stack as well as the caller parameters.
       The purpose of this method is to track down all the places in DIRAC that might
       not survive the change to JSON encoding.
 
+      Some methods are ignored:
+
+      * all the AccountingDB method: why ?
```

Comment:
I don't think there's any reason for that.

Reply:
yes there is a reason: the buckets are in tuples, and the Accouting uses DEncode to store data, not to send it over the network. So it just makes useless noise. I'll update the comment